bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,198 @@
+            "reveled": reveled_board,
+            "dm_msg": dm_msg,
+            "chat_msg": chat_msg
+        }
+
+    @staticmethod
+    def get_cords(value1: str, value2: str) -> typing.Tuple[int, int]:
+        """Take in 2 values for the cords and turn them into numbers"""
+        if value1.isnumeric():
+            return ord(value2.lower()) - 97, int(value1) - 1
+        else:
+            return ord(value1.lower()) - 97, int(value2) - 1
+
+    async def reload_board(self, ctx: commands.Context) -> None:
+        """Update both playing boards."""
+        game = self.games[ctx.author]
+        await game["dm_msg"].delete()
+        game["dm_msg"] = await ctx.author.send(self.format_for_discord(game["reveled"]))
+        await game["chat_msg"].edit(content=self.format_for_discord(game["reveled"]))
+
+    @commands.dm_only()
+    @commands.command(name="flag")
+    async def flag_command(self, ctx: commands.Context, *cords) -> None:
+        """Place multiple flags on the board"""
+        board = self.games[ctx.author]["reveled"]
+        for cord in cords:
+            x, y = self.get_cords(cord[0], cord[1])
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.reload_board(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author]
+        game["reveled"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":fire: You lost :fire: ")
+        await game["chat_msg"].channel.send(f":fire: {ctx.author.mention} just lost minesweeper :fire:")
+        del self.games[ctx.author]
+
+    async def won(self, ctx: commands.Context):
+        """The player won the game"""
+        game = self.games[ctx.author]
+        game["reveled"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":tada:  You won! :tada: ")
+        await game["chat_msg"].channel.send(f":tada: {ctx.author.mention} just won minesweeper :tada:")
+        del self.games[ctx.author]
+
+    def reveal_zeros(self, reveled: typing.List, board: typing.List, x: int, y: int) -> None:
+        """Used when a 0 is encountered to do a flood fill"""
+        for x_ in [x - 1, x, x + 1]:
+            for y_ in [y - 1, y, y + 1]:
+                if x_ == -1 or x_ == 10 or y_ == -1 or y_ == 10 or reveled[y_][x_] != "hidden":
+                    continue
+                reveled[y_][x_] = board[y_][x_]
+                if board[y_][x_] == 0:
+                    self.reveal_zeros(reveled, board, x_, y_)
+
+    async def reveal_one(self, ctx: commands.Context, reveled: typing.List, board: typing.List, x: int, y: int) -> None:
+        """Reveal one square."""
+        reveled[y][x] = board[y][x]
+        if board[y][x] == "bomb":
+            await self.lost(ctx)
+        elif board[y][x] == 0:
+            self.reveal_zeros(reveled, board, x, y)
+
+        # check if won
+        break_ = False
```

Comment:
Would it be better to use a function for this so you can return instead of double breaking?